src/main/java/codeu/controller/ProfileServlet.java

```diff
@@ -0,0 +1,178 @@
+  private MessageStore messageStore;
+
+  /**All the messages for a given ID */
+  private List <Message> messages;
+
+
+  /**All the conversations */
+  private List <Conversation> conversations;
+
+  /* Only messages that correspond to a given user ID */
+  private List <Message> realMessages;
+
+  /* List of all I.Ds for a given conversation   */
+  private List <UUID>conversationID;
+
+
+/**
+   * Set up state for handling profile-related requests. This method is only called when running in a
+   * server, not when running in a test.
+   */
+  @Override
+  public void init() throws ServletException {
+    super.init();
+    setUserStore(UserStore.getInstance());
+    setConversationStore(ConversationStore.getInstance());
+    setMessageStore(MessageStore.getInstance());
+  }
+  /**
+   * Sets the UserStore used by this servlet. This function provides a common setup method for use
+   * by the test framework or the servlet's init() function.
+   */
+  void setUserStore(UserStore userStore) {
+    this.userStore = userStore;
+  }
+  /**
+   * Sets the ConversationStore used by this servlet. This function provides a common setup method for use
+   * by the test framework or the servlet's init() function.
+   */
+  void setConversationStore(ConversationStore conversationStore) {
+    this.conversationStore = conversationStore;
+  }
+  /**
+   * Sets the MessageStore used by this servlet. This function provides a common setup method for use
+   * by the test framework or the servlet's init() function.
+   */
+  void setMessageStore(MessageStore messageStore) {
+    this.messageStore = messageStore;
+  }
+  /**
+   * This function fires when a user requests the /profile URL. It gets the ID of the user of the profile page from
+   * the URL and finds the corresponding about information.
+   * It then forwards to profile.jsp for rendering.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException {
+
+    String requestUrl = request.getRequestURI();
+    Matcher m = Pattern.compile("^/profile/(.+)").matcher(requestUrl);
+    if (!m.find()) {
+      System.out.println("wrong URL pattern: " + requestUrl);
+      return; // return error 
+    }
+    String extractedId = m.group(1);
+    System.out.println("extracted Id from URL (doGet): " + extractedId);
+    UUID profileId = UUID.fromString(extractedId);
+
+
+    String username = (String) request.getSession().getAttribute("user");
+    if (username == null) {
+      // User is not logged in, don't let them see a profile
+      System.out.println("Please login before viewing profiles");
+      response.sendRedirect("/login");
+      return;
+    }
+
+    User user = userStore.getUser(profileId); 
+
+    conversations = conversationStore.getAllConversations(); //list of conversations
+    for(Conversation c : conversations) {
+      conversationID.add(c.getId());  //list of I.Ds
+    }
```

Comment:
is this loop necessary? can't we just do 
`for (Conversation c : conversations)`
 `messages = messageStore.getMessagesInConversation(c.getId());`
  `.....`
instead of creating a new list of conversation id's?